### ISSUE TITLE
User is not prompted to install missing `nbconvert` dependency when exporting notebooks

### ISSUE THREAD

**Issue by david-waterworth** [microsoft/vscode-jupyter#5726]
When i try and convert from an ipynb to a py file, I get `ModuleNotFoundError: No module named 'nbconvert'`

## Environment data

-   VS Code version: 1.55.2
-   Jupyter Extension version (available under the Extensions sidebar): v2021.5.745244803
-   Python Extension version (available under the Extensions sidebar): v2021.4.765268190
-   OS (Windows | Mac | Linux distro) and version: Ubuntu 20.04
-   Python and/or Anaconda version: 3.7.6
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): pyenv virtualenv
-   Jupyter server running: Local

## Expected behaviour

Can convert a notebook to a script

## Actual behaviour

Exception trying to convert a notebook to a script

## Steps to reproduce:

Open any notebook, Export As "Python Script"

## Workaround

Manually install 'nbconvert'

> pip install nbconvert

I've attached the logs below, looks like vscode installed jupyter-core but not nbconvert

## Logs

```
Info 2021-05-03 15:03:40: Process Execution: > ~/.pyenv/versions/3.7.6/envs/warp-speed/bin/python ~/.vscode/extensions/ms-toolsai.jupyter-2021.5.745244803/pythonFiles/pyvsc-run-isolated.py jupyter nbconvert --version
> ~/.pyenv/versions/3.7.6/envs/warp-speed/bin/python ~/.vscode/extensions/ms-toolsai.jupyter-2021.5.745244803/pythonFiles/pyvsc-run-isolated.py jupyter nbconvert --version
Error 2021-05-03 15:03:40: Export failed [i [Error]: Traceback (most recent call last):
  File "/home/david/.local/bin/jupyter-nbconvert", line 7, in <module>
    from nbconvert.nbconvertapp import main
ModuleNotFoundError: No module named 'nbconvert'

	at ChildProcess.<anonymous> (/home/david/.vscode/extensions/ms-toolsai.jupyter-2021.5.745244803/out/client/extension.js:24:222504)
	at Object.onceWrapper (events.js:422:26)
	at ChildProcess.emit (events.js:315:20)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)] {
  category: 'unknown'
}
```

```
Collecting nbconvert
  Downloading nbconvert-6.0.7-py3-none-any.whl (552 kB)
     |████████████████████████████████| 552 kB 12.6 MB/s 
Requirement already satisfied: jupyter-core in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbconvert) (4.7.1)
Collecting defusedxml
  Downloading defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting jupyterlab-pygments
  Downloading jupyterlab_pygments-0.1.2-py2.py3-none-any.whl (4.6 kB)
Collecting nbformat>=4.4
  Downloading nbformat-5.1.3-py3-none-any.whl (178 kB)
     |████████████████████████████████| 178 kB 21.1 MB/s 
Collecting nbclient<0.6.0,>=0.5.0
  Downloading nbclient-0.5.3-py3-none-any.whl (82 kB)
     |████████████████████████████████| 82 kB 1.2 MB/s 
Collecting bleach
  Downloading bleach-3.3.0-py2.py3-none-any.whl (283 kB)
     |████████████████████████████████| 283 kB 21.5 MB/s 
Collecting entrypoints>=0.2.2
  Using cached entrypoints-0.3-py2.py3-none-any.whl (11 kB)
Requirement already satisfied: pygments>=2.4.1 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbconvert) (2.8.1)
Collecting pandocfilters>=1.4.1
  Downloading pandocfilters-1.4.3.tar.gz (16 kB)
Collecting testpath
  Downloading testpath-0.4.4-py2.py3-none-any.whl (163 kB)
     |████████████████████████████████| 163 kB 21.9 MB/s 
Requirement already satisfied: jinja2>=2.4 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbconvert) (2.11.3)
Collecting mistune<2,>=0.8.1
  Using cached mistune-0.8.4-py2.py3-none-any.whl (16 kB)
Requirement already satisfied: traitlets>=4.2 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbconvert) (5.0.5)
Requirement already satisfied: MarkupSafe>=0.23 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jinja2>=2.4->nbconvert) (1.1.1)
Requirement already satisfied: jupyter-client>=6.1.5 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert) (6.2.0)
Collecting async-generator
  Downloading async_generator-1.10-py3-none-any.whl (18 kB)
Requirement already satisfied: nest-asyncio in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert) (1.5.1)
Requirement already satisfied: tornado>=4.1 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert) (6.1)
Requirement already satisfied: python-dateutil>=2.1 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert) (2.8.1)
Requirement already satisfied: pyzmq>=13 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jupyter-client>=6.1.5->nbclient<0.6.0,>=0.5.0->nbconvert) (22.0.3)
Collecting jsonschema!=2.5.0,>=2.4
  Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
     |████████████████████████████████| 56 kB 7.0 MB/s 
Requirement already satisfied: ipython-genutils in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from nbformat>=4.4->nbconvert) (0.2.0)
Requirement already satisfied: attrs>=17.4.0 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (20.3.0)
Requirement already satisfied: importlib-metadata in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (3.10.1)
Collecting pyrsistent>=0.14.0
  Downloading pyrsistent-0.17.3.tar.gz (106 kB)
     |████████████████████████████████| 106 kB 21.2 MB/s 
Requirement already satisfied: six>=1.11.0 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (1.15.0)
Requirement already satisfied: setuptools in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (41.2.0)
Requirement already satisfied: packaging in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from bleach->nbconvert) (20.9)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Requirement already satisfied: typing-extensions>=3.6.4 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from importlib-metadata->jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (3.7.4.3)
Requirement already satisfied: zipp>=0.5 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from importlib-metadata->jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert) (3.4.1)
Requirement already satisfied: pyparsing>=2.0.2 in /home/david/.pyenv/versions/3.7.6/envs/warp-speed/lib/python3.7/site-packages (from packaging->bleach->nbconvert) (2.4.7)
Building wheels for collected packages: pandocfilters, pyrsistent
  Building wheel for pandocfilters (setup.py) ... done
  Created wheel for pandocfilters: filename=pandocfilters-1.4.3-py3-none-any.whl size=7992 sha256=2e53e1294fe91eabde439c4e2ffedc95081e3bf4d0e01367220b89cf8c4d352a
  Stored in directory: /home/david/.cache/pip/wheels/42/81/34/545dc2fbf0e9137811e901108d37fc04650e81d48f97078000
  Building wheel for pyrsistent (setup.py) ... done
  Created wheel for pyrsistent: filename=pyrsistent-0.17.3-cp37-cp37m-linux_x86_64.whl size=130115 sha256=913a74c3b7c6242cb6b0b4b0f3d4a02e790deae3b79725f5cdb114fc898d5f94
  Stored in directory: /home/david/.cache/pip/wheels/a5/52/bf/71258a1d7b3c8cbe1ee53f9314c6f65f20385481eaee573cc5
Successfully built pandocfilters pyrsistent
Installing collected packages: pyrsistent, jsonschema, webencodings, nbformat, async-generator, testpath, pandocfilters, nbclient, mistune, jupyterlab-pygments, entrypoints, defusedxml, bleach, nbconvert
Successfully installed async-generator-1.10 bleach-3.3.0 defusedxml-0.7.1 entrypoints-0.3 jsonschema-3.2.0 jupyterlab-pygments-0.1.2 mistune-0.8.4 nbclient-0.5.3 nbconvert-6.0.7 nbformat-5.1.3 pandocfilters-1.4.3 pyrsistent-0.17.3 testpath-0.4.4 webencodings-0.5.1
```

**Comment by joyceerhl** [comment]
@david-waterworth Thanks for the bug. Just to clarify:

1. Was nbconvert already installed in your current environment when you attempted the export?
2. Are the second set of logs the terminal output from you manually installing nbconvert yourself?
3. Did you get prompted to install nbconvert or did it simply fail with an error?

**Comment by joyceerhl** [comment]
We should prompt to install nbconvert if we're not already doing that.

**Comment by david-waterworth** [comment]
@joyceerhl, no nbconvert wasn't already installed. The second set of logs shows what was installed when I manually installed nbconvert. I've never installed nbconvert in the past, whenever I've started a notebook in a new virtualenv everything automatically installs (or at least I've not had an issue using export in a new environment without installing any additional notebook dependencies). It was only an error message, not prompt.
